### ISSUE TITLE
[Bug] MapboxLayer not appearing at all when using scripting API

### ISSUE THREAD

**Issue by csoltas** [visgl/deck.gl#7121]
### Description

I'm trying to interleave deck.gl layers and Mapbox layers using the scripting API. I used the [docs' example](https://deck.gl/gallery/mapbox-layer) as a starting point, but you'll notice even the example isn't working if you click that link...

- The deck layers (`ScatterplotLayer` and `ArcLayer`) don't appear.
- Mapbox's text label layers also don't appear.

I recreated the docs' example with a regular mapbox layer (not deck.gl layer) and it worked fine. I also recreated it without using the `MapboxLayer` class, i.e. not interleaving - also worked fine. The issue therefore seems specific to deck.gl's `MapboxLayer` class.

Note: I was able to get the Mapbox text labels to appear by removing the `beforeId?` argument from the `map.addLayer()` calls, but the deck.gl layers still aren't showing up. :(

Another note: I also read #6852, #6854 and #5872. I tried to do what they all did - explicitly create a `Deck` or `DeckGL` instance and give it a `ScatterplotLayer` with an id that matches my `MapboxLayer`'s id. This didn't solve the problem either (though it's possible I did something wrong).

### Flavors

- [ ] React
- [ ] Python/Jupyter notebook
- [X] MapboxLayer
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] DeckLayer/DeckRenderer for ArcGIS

### Expected Behavior

Something like [this image from the docs](https://deck.gl/docs/developer-guide/base-maps/using-with-mapbox) on interleaving layers:

<img width="1136" alt="Screen Shot 2022-07-04 at 11 43 20 AM" src="https://user-images.githubusercontent.com/6313691/177205202-df1383bf-2413-4886-ae2e-4f6d865b3700.png">

### Steps to Reproduce

Here is an example that shows the same dataset works when interleaved as a regular mapbox layer, but does not work when interleaved as a deck.gl `MapboxLayer` instance.
```
<html>
  <head>
    <title>Interleaving deck.gl with Mapbox Layers</title>
    <script src="https://unpkg.com/deck.gl@latest/dist.min.js"></script>
    <script src="https://api.tiles.mapbox.com/mapbox-gl-js/v1.4.0/mapbox-gl.js"></script>
    <link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.53.0/mapbox-gl.css' rel='stylesheet' />

  </head>

  <body style="margin:0"></body>

  <script type="text/javascript">

     mapboxgl.accessToken = "<key>";
      
     const map = new mapboxgl.Map({
         container: document.body,
         style: 'mapbox://styles/mapbox/light-v9',
         center: [-122.402, 37.79],
         zoom: 10,
         pitch: 0
     });
     
     const {MapboxLayer, GeoJsonLayer} = deck;
      
     map.on('load', () => {
       const firstLabelLayerId = map.getStyle().layers.find(layer => layer.type === 'symbol').id;
       
       // Deck.gl approach - this layer won't appear for some reason
       const urbanAreasLayer = new deck.MapboxLayer({
         id: "deckgl-urbanAreas",
         type: deck.GeoJsonLayer,
         data: 'https://docs.mapbox.com/mapbox-gl-js/assets/ne_50m_urban_areas.geojson',
         getFillColor: [255,0,136],
         opacity: 0.4
       });
       map.addLayer(urbanAreasLayer, firstLabelLayerId);

//         // Mapbox.gl approach - Works, if you comment out the "Deck.gl approach" lines above
//         map.addSource('urban-areas', {
//           'type': 'geojson',
//           'data': 'https://docs.mapbox.com/mapbox-gl-js/assets/ne_50m_urban_areas.geojson'
//         });
//         map.addLayer(
//           {
//             'id': 'urban-areas-fill',
//             'type': 'fill',
//             'source': 'urban-areas',
//             'layout': {},
//             'paint': {
//                 'fill-color': '#f08',
//                 'fill-opacity': 0.4
//             }
//           },
//           firstLabelLayerId
//         );
         
     });

  </script>
</html>
```

### Environment

- Framework version: v8.8 (scripting API using "@latest")
- Browser: Chrome Version 103.0.5060.53
- OS: macOS Monterey Version 12.4

### Logs

- `Uncaught TypeError: t.getPadding is not a function`
- `Uncaught TypeError: Cannot read properties of null (reading 'userData')`

<img width="545" alt="Screen Shot 2022-07-04 at 11 49 27 AM" src="https://user-images.githubusercontent.com/6313691/177205733-0707ed75-9cf7-4867-bd3f-fa327268366f.png">

**Comment by Pessimistress** [comment]
1.4.0 is at least 4 years old. You should try using a more recent version of Mapbox.

**Comment by Pessimistress** [comment]
1.4.0 is at least 4 years old. You should try using a more recent version of Mapbox.

**Comment by csoltas** [comment]
Yup, that was it. Looks like the example in the docs had the same issue.

**Comment by Pessimistress** [comment]
Yes the example needs to be updated.

It probably would also help if we add a minimal supported version check to the MapboxLayer.

**Comment by Pessimistress** [comment]
Example updated, along with a bug fix in v8.8.3.
